packages/apollo-collaboration-server/src/fileHandling/fileHandling.service.ts

```diff
@@ -503,4 +508,107 @@ export class FileHandlingService {
     this.logger.debug('GFF3 is not loaded in cache!')
     return false
   }
+
+  /**
+   * This method check that each line has unique id. If not it creates one for each line and overwrites the orignal file
+   * @param filenameWithPath - filename with path
+   */
+  async checkGFF3uniqueKey(filenameWithPath: string) {
+    const stringOfGFF3 = await fs.readFile(filenameWithPath, {
+      encoding: 'utf8',
+      flag: 'r',
+    })
+    this.logger.verbose(`Data read from file=${stringOfGFF3}`)
+
+    const arrayOfThings = gff.parseStringSync(stringOfGFF3, {
+      parseAll: true,
+    })
+    this.logger.debug(`Setting apollo_ids if required....`)
+
+    // Loop all lines and check if each line has 'apollo_id' property inside attributes
+    for (const entry of arrayOfThings) {
+      // Comment, Directive and FASTA -entries are not presented as an array
+      if (Array.isArray(entry)) {
+        for (const [key, val] of Object.entries(entry)) {
+          this.logger.verbose(
+            `GFF3Item: key=${JSON.stringify(key)}, value=${JSON.stringify(
+              val,
+            )}`,
+          )
+          if (val.hasOwnProperty('attributes')) {
+            const assignedVal = Object.assign(val)
+            // Let's add apollo_id to parent feature if it doesn't exist
+            if (!assignedVal.attributes.hasOwnProperty('apollo_id')) {
+              assignedVal.attributes.apollo_id = uuidv4()
+            }
+            // Check if there is also childFeatures in parent feature and it's not empty
+            if (
+              val.hasOwnProperty('child_features') &&
+              Object.keys(assignedVal.child_features).length > 0
+            ) {
+              // Let's add apollo_id to each child recursively
+              this.setApolloIdRecursively(assignedVal)
+            }
+          }
+        }
+      }
+    }
+
+    // Save into file
+    const { FILE_SEARCH_FOLDER, GFF3_DEFAULT_FILENAME_AT_STARTUP } = process.env
+    if (!FILE_SEARCH_FOLDER) {
+      throw new Error('No FILE_SEARCH_FOLDER found in .env file')
+    }
+    if (!GFF3_DEFAULT_FILENAME_AT_STARTUP) {
+      throw new Error('No GFF3_DEFAULT_FILENAME_AT_STARTUP found in .env file')
+    }
+    await fs.writeFile(
+      join(FILE_SEARCH_FOLDER, GFF3_DEFAULT_FILENAME_AT_STARTUP),
+      gff.formatSync(arrayOfThings),
+    )
+  }
+
+  /**
+   * Loop child features in parent feature and add apollo_id to each child's attribute
+   * @param parentFeature - Parent feature
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setApolloIdRecursively(parentFeature: any) {
```

Comment:
Should be `GFF3FeatureLineWithRefs` instead of `any`

```suggestion
  setApolloIdRecursively(parentFeature: GFF3FeatureLineWithRefs) {
```